### ISSUE TITLE
Why was this not submitted as a part of Configurable report itself?

### ISSUE THREAD

**Issue by mudrd8mz** [safatshahin-dev/moodle-local_configurable_api#1]
Why is this implemented as another plugin? Was there an attempt to talk to the maintainers of Configurable reports plugin and contribute it as a part of it instead?

**Comment by safatshahin** [comment]
Hi @mudrd8mz very good point. Before starting the build, I had four features I wanted to implement for this plugin:

1. get configurable report data via an API
2. push that data to another API
3. pass parameters to the report and get the relevant data
4. map the JSON data from moodle to the parameters of an external API and push the data

thinking of those features, I am relying on configurable for the base data, means query validation and stuffs like that. 
but every time I want to push any of those enhancements, I will have to get back to a conversation with the lead maintainer for that plugin and wait for their acceptance. From a developer point of view, it will be an easier path for me to enhance this plugin on my own pace.

Hope I could explain my point of making it an individual plugin.

**Comment by mudrd8mz** [comment]
>  I will have to get back to a conversation with the lead maintainer for that plugin and wait for their acceptance

Yes. That is what we call "co-operation" :-) Seriously though, I still believe it is worth trying at least. The Configurable Reports has been maintained by respected Moodle developers community members. The block is widely used and I can't stop thinking - why not to bring these nice features to all 12.000+ installations instead?

Can we please consider at least trying to contact @jleyva and @sarjona to see what they think about the idea? Chances are these will be specific features that would not fit into their plans with the plugin. Or they can admit they would not have time to check your patches. But I don't want to miss this opportunity.

Thank you for understanding.

**Comment by jleyva** [comment]
Hi there,

we've received dozens of contributions for configurable reports, most of them were integrated after a few minor iterations with the developers to fix things we felt required some adjustments. 

While I could understand that it would be easier to do everything by yourself, please consider that you are missing a wider audience having a separate plugin. As David said, the beauty of open-source is co-operation.

Feel free to reach anytime, we might be a bit slow responding but we'll do it!

Regards, Juan

**Comment by safatshahin** [comment]
@mudrd8mz, 100% agree with you, our community is built on collaboration and we should always try to keep that rolling.
@jleyva Thank you so much for getting into this, I will create an issue in the configurable report and also implement this plugin making a branch ahead of MOODLE_36_STABLE

**Comment by mudrd8mz** [comment]
Thank you guys! As said before - if there is any valid reason not to have it merged into the block itself, let's publush this as an add-on. I am happy that you'll try. Let me know if I can help with anything. Good luck!
